### PR TITLE
remove base64 encoding of ids

### PR DESCRIPTION
**Pull request by micbar**
## Change: Unify file IDs

We changed the file IDs to be consistent across all our APIs (WebDAV, LibreGraph, OCS). We removed the base64 encoding. Now they are formatted like `<storageID>!<opaqueID>`. They are using a reserved character ``!`` as a URL safe separator.

## Related Issue

ocis PR https://github.com/owncloud/ocis/pull/3185

## Examples

```bash
curl -L -X PROPFIND 'https://localhost:9200/dav/spaces/6afcb97a-e632-447d-b832-1334a8a9634f' \
-H 'Depth: infinity' \
-H 'Authorization: Basic YWRtaW46YWRtaW4='
```

```xml
<d:multistatus xmlns:s="http://sabredav.org/ns" xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
    <d:response>
        <d:href>/dav/spaces/6afcb97a-e632-447d-b832-1334a8a9634f/</d:href>
        <d:propstat>
            <d:prop>
                <oc:id>6afcb97a-e632-447d-b832-1334a8a9634f!6afcb97a-e632-447d-b832-1334a8a9634f</oc:id>
                <oc:fileid>6afcb97a-e632-447d-b832-1334a8a9634f!6afcb97a-e632-447d-b832-1334a8a9634f</oc:fileid>
                <d:getetag>&#34;55e95f454e48c0047d901686f4f7027c&#34;</d:getetag>
                <oc:permissions>RDNVCK</oc:permissions>
                <d:resourcetype>
                    <d:collection/>
                </d:resourcetype>
                <oc:size>2019472</oc:size>
                <d:getlastmodified>Tue, 15 Feb 2022 11:42:10 GMT</d:getlastmodified>
                <oc:favorite>0</oc:favorite>
            </d:prop>
            <d:status>HTTP/1.1 200 OK</d:status>
        </d:propstat>
    </d:response>
    <d:response>
        <d:href>/dav/spaces/6afcb97a-e632-447d-b832-1334a8a9634f/Space_Image.jpg</d:href>
        <d:propstat>
            <d:prop>
                <oc:id>6afcb97a-e632-447d-b832-1334a8a9634f!83a2c6ab-19f2-4f51-b3cb-41aef54ee5ab</oc:id>
                <oc:fileid>6afcb97a-e632-447d-b832-1334a8a9634f!83a2c6ab-19f2-4f51-b3cb-41aef54ee5ab</oc:fileid>
                <d:getetag>&#34;0440204b65664c18fa73a40cfe625815&#34;</d:getetag>
                <oc:permissions>RDNVW</oc:permissions>
                <d:resourcetype></d:resourcetype>
                <d:getcontentlength>2019367</d:getcontentlength>
                <d:getcontenttype>image/jpeg</d:getcontenttype>
                <d:getlastmodified>Tue, 15 Feb 2022 11:42:10 GMT</d:getlastmodified>
                <oc:checksums>
                    <oc:checksum>SHA1:439c4a6de0489288bb88914913fd1e5ece7d1764 MD5:2540f4e8f934296b617973f1a5cd5ae3 ADLER32:1c056526</oc:checksum>
                </oc:checksums>
                <oc:favorite>0</oc:favorite>
            </d:prop>
            <d:status>HTTP/1.1 200 OK</d:status>
        </d:propstat>
    </d:response>
    <d:response>
        <d:href>/dav/spaces/6afcb97a-e632-447d-b832-1334a8a9634f/NewFile5.md</d:href>
        <d:propstat>
            <d:prop>
                <oc:id>6afcb97a-e632-447d-b832-1334a8a9634f!44b8c8ba-8f57-452f-9b28-85b2058bdac8</oc:id>
                <oc:fileid>6afcb97a-e632-447d-b832-1334a8a9634f!44b8c8ba-8f57-452f-9b28-85b2058bdac8</oc:fileid>
                <d:getetag>&#34;6f35041017b213007453cd8d74ccb331&#34;</d:getetag>
                <oc:permissions>RDNVW</oc:permissions>
                <d:resourcetype></d:resourcetype>
                <d:getcontentlength>105</d:getcontentlength>
                <d:getcontenttype>text/markdown</d:getcontenttype>
                <d:getlastmodified>Tue, 15 Feb 2022 11:41:34 GMT</d:getlastmodified>
                <oc:checksums>
                    <oc:checksum>SHA1:081272bd25665a1a884fd50f3143220411c6e8a3 MD5:9bdb7fc99749fb9b2207358aec781742 ADLER32:dd6928f0</oc:checksum>
                </oc:checksums>
                <oc:favorite>0</oc:favorite>
            </d:prop>
            <d:status>HTTP/1.1 200 OK</d:status>
        </d:propstat>
    </d:response>
    <d:response>
        <d:href>/dav/spaces/6afcb97a-e632-447d-b832-1334a8a9634f/NewFolder/</d:href>
        <d:propstat>
            <d:prop>
                <oc:id>6afcb97a-e632-447d-b832-1334a8a9634f!dc4399c7-1fed-46e1-b81d-36c3db1c74ba</oc:id>
                <oc:fileid>6afcb97a-e632-447d-b832-1334a8a9634f!dc4399c7-1fed-46e1-b81d-36c3db1c74ba</oc:fileid>
                <d:getetag>&#34;0e0ec707e062c806e5e48faefcdca725&#34;</d:getetag>
                <oc:permissions>RDNVCK</oc:permissions>
                <d:resourcetype>
                    <d:collection/>
                </d:resourcetype>
                <oc:size>0</oc:size>
                <d:getlastmodified>Tue, 15 Feb 2022 11:40:51 GMT</d:getlastmodified>
                <oc:favorite>0</oc:favorite>
            </d:prop>
            <d:status>HTTP/1.1 200 OK</d:status>
        </d:propstat>
    </d:response>
</d:multistatus>
```

### CODE REVIEW

**Comment by phil-davis** [comment]
Note: core PR https://github.com/owncloud/core/pull/39751 added the code to find the personal space id of a user. It reverse-engineers the base64-encoding and expects the ":" between parts. I expect that we need to adjust that code, put it in a branch and PR to core master, then point to that from this PR so that tests will run better.

File: pkg/utils/resourceid/owncloud.go [hunk 1]
```diff
@@ -73,5 +67,5 @@ func OwnCloudResourceIDWrap(r *provider.ResourceId) string {
 // - url safe, because the id might be used in a url, eg. the /dav/meta nodes
 // which is why we base64 encode it
```
Comment (phil-davis) [inline pkg/utils/resourceid/owncloud.go:68]: These comment lines will need updating or removing

**Comment by micbar** [comment]
@phil-davis Can we create a branch in oc10 where we change the discovery? We need to know that my changes here do not break stuff.

**Comment by phil-davis** [comment]
> @phil-davis Can we create a branch in oc10 where we change the discovery? We need to know that my changes here do not break stuff.

https://github.com/owncloud/core/pull/39793 core branch `update-getPersonalSpaceIdForUser` commit id `b8e7eb936d6648db43eb59ce171f10e14af90320`

Put that in `.drone.env` and see what happens to CI.

File: pkg/utils/resourceid/owncloud.go (hunk 1 above)
Comment (micbar) [reply pkg/utils/resourceid/owncloud.go]: done

**Comment by micbar** [comment]
@phil-davis I broke some move tests. Is that a problem with the expected failures or are they real failures?

**Comment by phil-davis** [comment]
> @phil-davis I broke some move tests. Is that a problem with the expected failures or are they real failures?

Those are core test scenario changes that are in core master and need a commit id bump, and the line numbers... sorting out. I will see if @kiranparajuli589 has time to do that today, and then you will be able to rebase to fix that.

**Comment by kiranparajuli589** [comment]
> > @phil-davis I broke some move tests. Is that a problem with the expected failures or are they real failures?
> 
> Those are core test scenario changes that are in core master and need a commit id bump, and the line numbers... sorting out. I will see if @kiranparajuli589 has time to do that today, and then you will be able to rebase to fix that.

@phil-davis @micbar https://github.com/cs3org/reva/pull/2545 is created for bumping.

**Comment by phil-davis** [comment]
> Should we be using the commit id from the branch update-getPersonalSpaceIdForUser?

no, just do an ordinary commit id bump in a separate PR. Then this PR can be rebased.

**Comment by phil-davis** [comment]
PR #2545 has been merged and bumped the core commit id to the current tip of master.

I rebased core PR https://github.com/owncloud/core/pull/39793 and it is now core cranch `update-getPersonalSpaceIdForUser` with commit `10f3f9aaf4bdd80e408b055e467a167f1cb86228`

@micbar you can rebase this PR, and put the above in `.drone.env` and maybe it will pass.

**Review by phil-davis — APPROVED** [review]

File: changelog/unreleased/remove-base64-encoding-of-ids.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Change: Do not encode webDAV ids to base64
+
+We removed the encoding of the IDs and use the format <storageID>!<opaqueID> with a delimiter. The used delimiter is url safe anc belongs to the reserved keys.
```
Comment (butonic) [inline changelog/unreleased/remove-base64-encoding-of-ids.md:3]:
```suggestion
We removed the base64 encoding of the IDs and use the format <storageID>!<opaqueID> with a `!` delimiter. As a reserved delimiter it is URL safe. The IDs will be XML and JSON encoded as necessary.
```

**Review by butonic — APPROVED** [review]

**Pull request merged**